### ISSUE TITLE
[WPStarter 3] any reason why wecodemore/wp-downloader doesn't work?

### ISSUE THREAD

**Issue by soderlind** [wecodemore/wpstarter#88]
**Describe the bug**
Running `composer install` fails with the following error message:

> No WordPress version found.
  To skip WP requirement check, set 'require-wp' to
  false in WP Starter configuration in composer.json.
  WP Starter failed.

**To Reproduce**

Run `composer install` on the following composer.json:
```json
{
	"name": "dss-web/nettsteder",
	"type": "project",
	"description": "nettsteder.regjeringen.no",
	"license": "proprietary",
	"require": {
		"composer/installers": "^1.6",
		"wecodemore/wpstarter": "^3",
		"wecodemore/wp-downloader": "^1.0"
	},
	"minimum-stability": "dev",
	"extra": {
		"installer-paths": {
			"public/wp-content/plugins/{$name}": [
				"type:wordpress-plugin"
			],
			"public/wp-content/mu-plugins/{$name}": [
				"type:wordpress-muplugin"
			],
			"public/wp-content/themes/{$name}": [
				"type:wordpress-theme"
			],
			"public/wp-content/{$name}": [
				"type:wordpress-dropin"
			]
		},
		"wordpress-content-dir": "public/wp-content",
		"wordpress-install-dir": "public/wp",
		"wp-downloader": {
			"version": "^5.0",
			"no-content": true,
			"target-dir": "public/wp"
		}
	},
	"repositories": [
		{
			"type": "composer",
			"url": "https://wpackagist.org"
		}
	]
}
```

**Expected behavior**

WP Starter recognise that WordPress is installed 

**Additional context**
I prefer using wp-dowloader since I can ask for the latest version (`"version": "^5.0"`) and `no-content`.

**Comment by soderlind** [comment]
#89 should give me what I want so I'm closing this issue

**Comment by gmazzap** [comment]
Hi @soderlind 

Thanks for your reporting.

even if moving content can give you what you want, I want to make sure that v3 works with a WordPress source that is not a Composer package, so I'll re-open it.

I'll try to reproduce and fix as soon as I can.

**Comment by gmazzap** [comment]
Hi again @soderlind 

it seems there's nothing to do here :)

**TLDR;** as the error message was telling you you need to setup a configuration value for WP Starter 3:

```json
{
    "extra": {
        "wpstarter": {
                "require-wp": false
        }
    }
}
```

Basically, WP Starter 2 required WordPress from a specific Composer package (the one by John P. Bloch). 

WP Starter 3 does not require WordPress, letting users choose where and how to get WordPress.

Normally, WP Starter 3 tries to ensure WordPress version, by looking at any Composer packages of type `wordpress-core` among declared dependencies.

At the moment, there are only 2 of these packages at packagist.org (https://packagist.org/?query=wordpress-core&type=wordpress-core) but more could appear in future an nothing prevents people to use custom packages of that type, if they want.

However, when not requireing WordPress from any package, WP Starter 3 will not be able to recognize the installed WordPress version so it emits the error... **unless** you tell it to STFU and don't care about WordPress at all, because you know what you're doing.

To do that, you need to setup a config `require-wp: false` in WP Starter config and shown above.

Sorry if the error message was not clear enough. I will try to make this more clear in the documentation.

**Comment by soderlind** [comment]
np, I'm reading your doc and I'm learning more and more. I understand now that I should have set `require-wp` to false.

btw, I really like how WP Starter (3) works, and all the features you've added (and not added).

**Comment by gmazzap** [comment]
Thanks @soderlind :)

And, just for posterity, the actual packagist.org search for `wordpress-core` packges include at least 15 packages right now: https://packagist.org/?type=wordpress-core&tags=wordpress~wp

Will keep open, so I remeber to try to amke docs more clear.

**Comment by gmazzap** [comment]
Added explicit documentation at the end of this section: https://github.com/wecodemore/wpstarter/blob/dev/docs/03-WordPress-Integration.md#include-wordpress-in-the-project

> Please note: if WordPress is not installed as Composer package (and that the case for wp-downloader) then the WP Starter setting { "require-wp": false } must be set or WP starter will look for WordPress among packages and will fail not finding it.

Closing.
